riscv/insns/lpad.h

```diff
@@ -0,0 +1,9 @@
+if (p->extension_enabled(EXT_ZICFILP)) {
+  if (ZICFILP_READ_XLPE(STATE.v, STATE.prv) && STATE.elp == elp_t::LP_EXPECTED) {
+    software_check(pc % 4 == 0 &&
+                   ((READ_REG(7) >> 12) == insn.u_imm() || insn.u_imm() == 0),
```

Comment:
This is giving a warning from my (old) compiler:
```
.../riscv-isa-sim/riscv/insns/lpad.h: In function ‘reg_t logged_rv32e_lpad(processor_t*, insn_t, reg_t)’:
.../riscv-isa-sim/riscv/insns/lpad.h:4:41: warning: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘int64_t’ {aka ‘long int’} [-Wsign-compare]
    ((READ_REG(7) >> 12) == insn.u_imm() || insn.u_imm() == 0),
     ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
```

Reply:
I did a test run with my compiler (clang 17) and it did not issue a warning. 
Still, it's better to make the sign coherent and avoid possible warnings.
Thanks for the note :)

Fixed.